scipy/optimize/_qap.py

```diff
@@ -0,0 +1,413 @@
+import numpy as np
+import math
+from . import linear_sum_assignment, minimize_scalar, OptimizeResult
+
+
+def quadratic_assignment(
+    cost_matrix,
+    dist_matrix,
+    seed=np.array([[], []]).T,
+    maximize=False,
+    n_init=1,
+    init_method="barycenter",
+    max_iter=30,
+    shuffle_input=True,
+    eps=0.05,
+):
+    """
+    Solve the quadratic assignment problem
+
+    This class solves the Quadratic Assignment Problem and the Graph
+    Matching Problem (QAP) through an implementation of the Fast
+    Approximate QAP Algorithm (FAQ) (these two problems are the same up
+    to a sign change) [1]_.
+
+    This algorithm can be thought of as finding an alignment of the
+    vertices of two graphs which minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements. The option to add seeds
+    (known vertex correspondence between some nodes) is also available
+    [2]_.
+
+
+    Parameters
+    ----------
+    cost_matrix : 2d-array, square
+        A square adjacency matrix
+
+    dist_matrix : 2d-array, square
+        A square adjacency matrix
+
+    seed : 2d-array, optional, (default = np.array([[], []]).T)
```

Comment:
"seed" is often used to mean a guess for an iterative algorithm that gets refined. These are fixed. Is there any other name used in the literature instead of seed that would avoid any chance of ambiguity?

Also, could the algorithm handle / might the user want to specify _guesses_?